### PR TITLE
Test preimage existence on range of hashes

### LINKED ISSUE #337: Test Preimage Existence on a Range of Hash Images
This is an integration test that tests a bunch of other features end-to-end:
    - uses IntegerRangeQuantifier and the PreimageExistenceDecider to make decisions on ranges of preimages
    - uses a PropertyFactory which contains a list of `hashImages: bytes32[]` and constructs a PreimageExists Property with input as `hashImages[result]` via `for (result of IntegerRangeQuantifier.getAllSuchThat(...).results)`

### PR DESCRIPTION
**Pull request by willmeister**
## Description
Integration test for HashExistenceDecider over a range, using ForAllSuchThatDecider and IntegerRangeQuantifier to demonstrate.

### Fixes
- Fixes #337 

## Contributing Agreement

- [x] I have read and understood the [Plasma Group Contributing Guide and Code of Conduct](https://github.com/plasma-group/pigi/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

**Comment by willmeister** [comment]
Note: This PR's branch is off of [this PR](https://github.com/plasma-group/pigi/pull/377). It will be way easier to review this PR once the other one is merged.

File: packages/core/test/app/ovm/examples/preimage-existence-on-multiple-hashes.spec.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      }
+
+      const forAllInput: ForAllSuchThatInput = {
+        quantifier: rangeQuantifier,
+        quantifierParameters: { start: 2, end: 8 },
+        witnessFactory,
+        propertyFactory,
+      }
+
+      try {
```
Comment (ben-chain) [inline packages/core/test/app/ovm/examples/preimage-existence-on-multiple-hashes.spec.ts]: In the past I've used a `.should.throw(new Error('asdf'))`--just curious if you've seen this or whether there's an advantage to this approach :)

Comment (willmeister) [reply packages/core/test/app/ovm/examples/preimage-existence-on-multiple-hashes.spec.ts]: Haven't found a great way to do that with async calls, but maybe I can just wrap it in parens? `(await forAllDecider.decide(forAllInput, undefined).should.throw...`

**Comment by willmeister** [comment]
Included in #382

**Pull request closed without merging**